include/deal.II/grid/filtered_iterator.h

```diff
@@ -960,10 +960,8 @@ FilteredIterator<BaseIterator> &
 FilteredIterator<BaseIterator>::
 operator = (const FilteredIterator &fi)
 {
-  Assert ((fi.state() != IteratorState::valid) || (*predicate)(fi),
-          ExcInvalidElement(fi));
-  BaseIterator::operator = (fi);
-  return *this;
+  const BaseIterator &bi = fi;
+  return operator = (bi);
```

Comment:
Hm, ok.

If I saw the code you have, I'd probably be tempted to just change it back to its original form. Do you have a test that one could add to make sure we don't accidentally change things back?